docs/proposal/2022-09-28-event-manager-extensibility.md

```diff
@@ -0,0 +1,302 @@
+on RPC and the integration is so simple. 
+
+API folder contains `source` and `executor` folders to have more meaningful package names once we generate actual implementation of proto files. Apart from proto files, 
+you can see base gRPC server and client implementations in `grpc.go` which is shared among plugins. `interface.go` contains the actual API to expose to consumers. Finally,
+`plugin.go` is for plugin definition which is a simple struct for Hashicorp's Go plugin system.
+
+### Plugins Folder
+In this folder, you can see examples `kubectl` and `Kubernetes`. Each plugin has its own dedicated folders which are typically go module projects. 
+Each plugin folder contains a simple Go file which contains the actual business logic of plugin. You can also see a `Makefile` to manage the build
+process of that plugin. They might not be a simple Go program, that's why there is no unified `Makefile` that builds any Go project, it can contain
+customer specific build flows. For `kubectl` and `Kubernetes`, they are same for now. 
+
+### How to build plugins?
+`plugins` folder only contains the source code of the plugins, and consumer (Botkube or playground project on our case), needs the executable versions
+of those plugin to call via Hashicorp's plugin package. There is a Github Action for plugin release flow which you can see [here](https://github.com/huseyinbabal/Botkube-plugins/blob/main/.github/workflows/release.yaml). 
+In this flow, the only thing you need to do is pushing a new tag in a format `<plugin_name>/<version>` like `Kubernetes/v1.0.7`. This will build `Kubernetes` plugin, create a release with version
+`v1.0.7` and go binary executable will be attached to release artifacts. Notice that, those artifacts are used in consumer side to be used as parameter to Hashicorp's Go Plugin.
+
+### How can I know the metadata of plugins?
+In `Botkube-plugins` project, we have a plugin index file as you can also see it [here](https://github.com/huseyinbabal/Botkube-plugins/blob/main/index.json). That contains basic metadata of each plugin, and notice that they are managed manually
+right now, we can automate and do find/replace whenever there is a change to plugins. Here is an example;
+```json
+[
+  {
+    "name": "Kubernetes",
+    "type": "source",
+    "description": "Kubernetes source plugin to consume events from Kubernetes Events API.",
+    "version": "v1.0.9"
+  },
+  {
+    "name": "kubectl",
+    "type": "executor",
+    "description": "Kubectl executor plugin to handle given command in configured k8s cluster.",
+    "version": "v1.0.9"
+  }
+]
+```
+
+### How can I provide plugin configuration?
+If you use terraform before, they use a notation like `<pluginname>_...` to pass settings to specific plugin. So, in Botkube, if somebody provides `KUBERNETES_CONFIG_PATH=something`, the `kubernetes` plugin executable can be run as follows.
```

Comment:
probably will be good to also require a prefix `EXECUTOR_` and `SOURCE_` to have dedicated namespaces in case of plugin name conflict.